app/wizards/schools/assign_existing_mentor_wizard/lead_provider_step.rb

```diff
@@ -24,7 +26,48 @@ def persist
           author: wizard.author
         ).assign!
 
-        # TODO: Update the LP if there is a confirmed partnership, if not add an EOI
+        create_mentor_training_period!
+      end
+
+      def create_mentor_training_period!
```

Comment:
Suggestion to rename this to something like: `create_mentor_training_period_and_add_event!`

To make it clear an event is being added too?